pkg/gitreceive/build.go

```diff
@@ -0,0 +1,295 @@
+	}
+
+	makeBucketCmd := mcCmd(configDir, "mb", fmt.Sprintf("%s://%s:%s/git", storage.schema(), storage.host(), storage.port()))
+	// Don't look for errors here. Buckets may already exist
+	// https://github.com/deis/builder/issues/80 will eliminate this distaste
+	run(makeBucketCmd)
+
+	cpCmd := mcCmd(configDir, "cp", fmt.Sprintf("%s.tar.gz", appName), tarURL)
+	cpCmd.Dir = repoDir
+	if err := run(cpCmd); err != nil {
+		return fmt.Errorf("copying %s.tar.gz to %s (%s)", appName, tarURL, err)
+	}
+
+	log.Info("Starting build")
+	log.Debug("Starting pod %s", buildPodName)
+	kCreateCmd := exec.Command(
+		"kubectl",
+		fmt.Sprintf("--namespace=%s", conf.PodNamespace),
+		"create",
+		"-f",
+		finalManifestFileLocation,
+	)
+	if log.IsDebugging {
+		kCreateCmd.Stdout = os.Stdout
+	}
+	kCreateCmd.Stderr = os.Stderr
+	if err := run(kCreateCmd); err != nil {
+		return fmt.Errorf("creating builder pod (%s)", err)
+	}
+
+	// poll kubectl every 100ms to determine when the build pod is running
+	// TODO: use the k8s client and watch the event stream instead (https://github.com/deis/builder/issues/65)
+	for {
+		cmd := kGetCmd(conf.PodNamespace, buildPodName)
+		var out bytes.Buffer
+		cmd.Stdout = &out
+		// ignore errors
+		run(cmd)
+		outStr := string(out.Bytes())
+		if strings.Contains(outStr, "phase: Running") {
+			break
+		} else if strings.Contains(outStr, "phase: Failed") {
+			return fmt.Errorf("build pod %s entered phase: Failed", buildPodName)
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	// get logs from the builder pod
+	kLogsCmd := exec.Command(
+		"kubectl",
+		fmt.Sprintf("--namespace=%s", conf.PodNamespace),
+		"logs",
+		"-f",
+		buildPodName,
+	)
+	kLogsCmd.Stdout = os.Stdout
+	if err := run(kLogsCmd); err != nil {
+		return fmt.Errorf("running %s to get builder logs (%s)", strings.Join(kLogsCmd.Args, " "), err)
+	}
+
+	// poll the s3 server to ensure the slug exists
+	for {
+		// for now, assume the error indicates that the slug wasn't there, nothing else
+		// TODO: implement https://github.com/deis/builder/issues/80, which will clean this up siginficantly
+		lsCmd := mcCmd(configDir, "ls", pushURL)
+		if err := run(lsCmd); err == nil {
+			break
+		}
+	}
+
+	log.Info("Build complete.")
+
+	log.Info("Launching...")
+
+	buildHook := &pkg.BuildHook{
+		Sha:         gitSha,
+		ReceiveUser: conf.Username,
+		ReceiveRepo: appName,
+		Image:       appName,
+		Procfile:    procType,
+	}
+	if !usingDockerfile {
+		buildHook.Dockerfile = ""
+	} else {
+		buildHook.Dockerfile = "true"
+		buildHook.Image = imageName
+	}
+	buildHookResp, err := publishRelease(conf, builderKey, buildHook)
+	if err != nil {
+		return fmt.Errorf("publishing release (%s)", err)
+	}
+	release, ok := buildHookResp.Release["version"]
+	if !ok {
+		return fmt.Errorf("No release returned from Deis controller")
+	}
+
+	log.Info("Done, %s:v%d deployed to Deis", appName, release)
+	log.Info("Use 'deis open' to view this application in your browser")
```

Comment:
I believe the old layout was

```
done, ...

use deis open...

to learn more...
```

I'm going to test manually, but wouldn't the output now be

```
done, ...
use deis open...
to learn more...
```

Since there's a lack of extra newline characters? I know our BDFL was quite stingy on the format